src/core/mac/mac.cpp

```diff
@@ -1952,6 +1978,10 @@ void Mac::HandleReceivedFrame(RxFrame *aFrame, Error aError)
         break;
     }
 
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+    SuccessOrExit(error = HandleWakeupFrame(*aFrame));
```

Comment:
Is there a reason not to move this under the following switch statement under `case Frame::kTypeMultipurpose`?

Reply:
Hi, this comes from the time in which discussion was open to use other types than multipurpose as wake-up frames.

Reply:
Moved now.